src/sardana/pool/poolbasegroup.py

```diff
@@ -358,6 +358,8 @@ def stop(self):
             self.debug("Stopping %s %s", ctrl.name,
                        [e.name for e in elements])
             try:
+                for el in elements:
```

Comment:
I would move all the loop before of the try..except. Here we are just setting an internal flag, nothing can go wrong.

Reply:
Moved